tests/e2e/isbservice.go

```diff
@@ -229,3 +232,145 @@ func startISBServiceRolloutWatches() {
+
+	isbServiceRolloutSpec := createISBServiceRolloutSpec(name, Namespace, isbServiceSpec)
+	_, err := isbServiceRolloutClient.Create(ctx, isbServiceRolloutSpec, metav1.CreateOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying that the ISBServiceRollout was created")
+	Eventually(func() error {
+		_, err := isbServiceRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		return err
+	}, testTimeout, testPollingInterval).Should(Succeed())
+
+	verifyISBSvcRolloutReady(name)
+
+	verifyISBSvcReady(Namespace, name, 3)
+
+}
+
+func createISBServiceRolloutSpec(name, namespace string, isbServiceSpec numaflowv1.InterStepBufferServiceSpec) *apiv1.ISBServiceRollout {
+
+	rawSpec, err := json.Marshal(isbServiceSpec)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	isbServiceRollout := &apiv1.ISBServiceRollout{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "numaplane.numaproj.io/v1alpha1",
+			Kind:       "ISBServiceRollout",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: apiv1.InterStepBufferService{
+				Spec: runtime.RawExtension{
+					Raw: rawSpec,
+				},
+			},
+		},
+	}
+
+	return isbServiceRollout
+
+}
+
+// delete ISBServiceRollout and verify deletion
+func deleteISBServiceRollout(name string) {
+	document("Deleting ISBServiceRollout")
+	err := isbServiceRolloutClient.Delete(ctx, name, metav1.DeleteOptions{})
+	Expect(err).ShouldNot(HaveOccurred())
+
+	document("Verifying ISBServiceRollout deletion")
+	Eventually(func() bool {
+		_, err := isbServiceRolloutClient.Get(ctx, name, metav1.GetOptions{})
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				Fail("An unexpected error occurred when fetching the ISBServiceRollout: " + err.Error())
+			}
+			return false
+		}
+		return true
+	}).WithTimeout(testTimeout).Should(BeFalse(), "The ISBServiceRollout should have been deleted but it was found.")
+
+	document("Verifying ISBService deletion")
+	Eventually(func() bool {
+		list, err := dynamicClient.Resource(getGVRForISBService()).Namespace(Namespace).List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return false
+		}
+		if len(list.Items) == 0 {
+			return true
+		}
+		return false
+	}).WithTimeout(testTimeout).Should(BeTrue(), "The ISBService should have been deleted but it was found.")
+}
+
+func updateISBServiceRollout(isbServiceRolloutName, pipelineRolloutName string, newSpec numaflowv1.InterStepBufferServiceSpec, f func(numaflowv1.InterStepBufferServiceSpec) bool, dataLoss bool) {
+
+	rawSpec, err := json.Marshal(newSpec)
+	Expect(err).ShouldNot(HaveOccurred())
+
+	updateISBServiceRolloutInK8S(isbServiceRolloutName, func(rollout apiv1.ISBServiceRollout) (apiv1.ISBServiceRollout, error) {
+		rollout.Spec.InterStepBufferService.Spec.Raw = rawSpec
+		return rollout, nil
+	})
+
+	if upgradeStrategy == config.PPNDStrategyID && dataLoss {
```

Comment:
So, I guess `upgradeStrategy` is a global variable that the user will need to set for any test? That seems good I think.

Reply:
I believe it should work for any valid strategy or if it is empty since we're using the `IsValid()` function to verify that it is valid.

Reply:
But it looks like [IsValid(](https://github.com/numaproj/numaplane/blob/main/tests/e2e/functional_test.go#L194)) is only being called for the functional test. Perhaps we should call it from the common before suite logic?

Reply:
Sounds good, will move that over so that it should run for all suites.